FestivalPlanner/loader/views.py

```diff
@@ -66,10 +67,17 @@ class TheaterDataInterfaceView(LoginRequiredMixin, FormView):
     def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
         self.action = None
+        self.action_cookie = Cookie('action')
+
+    def setup(self, request, *args, **kwargs):
+        super().setup(request, *args, **kwargs)
+        session = request.session
+        self.action_cookie.init_cookie(session)
+        self.action = self.action_cookie.get(session)
 
     def dispatch(self, request, *args, **kwargs):
-        if 'action' in request.GET:
-            self.action = request.GET['action']
+        self.action_cookie.handle_request(request, self.action)
+        self.action = self.action_cookie.get(request.session)
```

Comment:
To me, this approach seems like hopping on two legs.
Please consider to make this simpler.

Reply:
Thanks for letting me look at his again!
It's much more simple now (done).